### PR TITLE
Fixed location menu ordering and added rendering for barrier station options

### LINKED ISSUE #752: Z order is wrong on Enter/Don't Enter dialogue
The frame around the image is behind the main textbox, and the selection highlight is in front of the text

### PR DESCRIPTION
**Pull request by theflyingzamboni**
Fixes #752

### CODE REVIEW

File: src/main/java/legend/game/wmap/WMap.java [hunk 1]
```diff
@@ -4951,30 +4951,34 @@ private void initializeLocationMenuTextHighlightEffects() {
...
     }
 
+    if(this.placeImage != null) {
+      this.placeImage.delete();
+      this.placeImage = null;
+    }
+
     if(this.mapState_800c6798.submapCut_c8 == 999) { // Going to a different region
+      if(this.destPlaceNames != null) {
+        this.destPlaceNames = null;
```
Comment (theflyingzamboni) [inline src/main/java/legend/game/wmap/WMap.java:4975]: Do I need to delete each object individually, or will nulling the array allow everything to be released and garbage collected?

File: src/main/java/legend/game/wmap/WMap.java [hunk 2]
```diff
@@ -5107,18 +5120,20 @@ private void handleMapTransitions() {
...
-          final int sp3c = this.mapState_800c6798.submapScene_ca & 0xf;
+          final int dest1 = this.mapState_800c6798.submapScene_ca >>> 4 & 0xffff;
+          final int dest2 = this.mapState_800c6798.submapScene_ca & 0xf;
 
           this.textTransforms.transfer.set(240.0f, 164.0f, textZ_800bdf00.get() * 4.0f);
           RENDERER.queueOrthoOverlayModel(this.dontEnter, this.textTransforms);
-
-          this.renderCenteredShadowedText(regions_800f01ec.get(sp38).deref(), 240, 182, TextColour.WHITE, 0);
-          this.renderCenteredShadowedText(regions_800f01ec.get(sp3c).deref(), 240, 200, TextColour.WHITE, 0);
+          this.textTransforms.transfer.setComponent(1, 182.0f);
```
Comment (LordMonoxide) [inline src/main/java/legend/game/wmap/WMap.java:5133]: You can just do `.y = 182.0f`

File: src/main/java/legend/game/wmap/WMap.java [hunk 3]
```diff
@@ -5107,18 +5120,20 @@ private void handleMapTransitions() {
...
+          RENDERER.queueOrthoOverlayModel(this.destPlaceNames[dest1], this.textTransforms);
+          this.textTransforms.transfer.setComponent(1, 200.0f);
```
Comment (LordMonoxide) [inline src/main/java/legend/game/wmap/WMap.java:5135]: Same here

File: src/main/java/legend/game/wmap/WMap.java (hunk 1 above)
Comment (LordMonoxide) [reply src/main/java/legend/game/wmap/WMap.java:4975]: They all need to be deleted from the GPU memory or it'll cause a GPU memory leak

File: src/main/java/legend/game/wmap/WMap.java [hunk 4]
```diff
@@ -5077,26 +5081,35 @@ private void handleMapTransitions() {
...
+            .build();
+
           this.dontEnter = new TextBuilder()
             .text("Don't enter")
             .centred()
             .shadowed()
             .build();
 
           if(this.mapState_800c6798.submapCut_c8 == 999) { // Going to a different region
+            this.destPlaceNames = new TextObj[regions_800f01ec.length()];
```
Comment (LordMonoxide) [inline src/main/java/legend/game/wmap/WMap.java:5098]: Do we need to upload renderables for every region to the GPU? Or do we know which region we should be displaying?

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

**Review by LordMonoxide — APPROVED** [review]

**Pull request merged**